episodes/03-explore-data.Rmd

```diff
@@ -0,0 +1,344 @@
+# Package names
+packages <- c("tidyverse", "here")
+
+# Packages loading
+invisible(lapply(packages, library, character.only = TRUE))
+
+```
+
+:::::::::::::::::::::::::::::::::::::: questions 
+
+- What is a data frame?
+- How can I read data in R?
+- How can I get basic summary information about my data set?
+- How can I select specific rows and/or columns from a data frame?
+- How can I combine multiple commands into a single command?
+- How can I create new columns or remove existing columns from a data frame?
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::: objectives
+
+- Describe what a data frame is.
+- Load external data from a .csv file into a data frame.
+- Summarize the contents of a data frame.
+- Select certain columns in a data frame with the dplyr function select.
+- Select certain rows in a data frame according to filtering conditions with the dplyr function filter.
+- Link the output of one dplyr function to the input of another function with the ‘pipe’ operator %>%.
+- Add new columns to a data frame that are functions of existing columns with mutate.
+- Use the split-apply-combine concept for data analysis.
+- Use summarize, group_by, and count to split a data frame into groups of observations, apply a summary statistics for each group, and then combine the results.
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+
+# [Exploring Data frames](https://datacarpentry.org/r-intro-geospatial/04-data-structures-part2/index.html)
+
+Now we turn to the bread-and-butter of working with `R`: working with tabular data. In `R` data are stored in a data structure called **data frames**.  
+
+A data frame is a representation of data in the format of a **table** where the columns are **vectors** that all have the **same length**. 
+
+
+Because columns are vectors, each column must contain a **single type of data** (e.g., characters, numeric, factors). 
+For example, here is a figure depicting a data frame comprising a numeric, a character, and a logical vector.
+
+![](assets/img/data-frame.svg)
+<font size="3">*Source*:[Data Carpentry R for Social Scientists ](https://datacarpentry.org/r-socialsci/02-starting-with-data/index.html#what-are-data-frames-and-tibbles)</font>
+
+
+## Reading data
+
+`read.csv()` is a function used to read coma separated data files (`.csv` format)). There are other functions for files separated with other delimiters. 
+We're gonna read in the `gapminder` data set with information about countries' size, GDP and average life expectancy in different years.
+
+```{r reading-data}
+gapminder <- read.csv("data/gapminder_data.csv")
+
+```
+
+## Exploring dataset
+Let’s investigate the `gapminder` data frame a bit; the first thing we should always do is check out what the data looks like.
+
+It is important to see if all the variables (columns) have the data type that we require. Otherwise we can run into trouble.
```

Comment:
'Otherwise we can run into trouble' might be a bit vague. We could give an example of an issue we can encounter if we do not check the data types (e.g.,  trying to calculate the min, max, or mean of the variable that has been wrongly encoded as a character vector). However, this explanation might be too obvious.

Reply:
```suggestion
It is important to see if all the variables (columns) have the data type that we require. For instance, a column might have numbers stored as characters, which would not allow us to make calculations with those numbers.
```

Reply:
Good suggestion, not obvious in this context.